app/src/org/commcare/recovery/measures/ExecuteRecoveryMeasuresPresenter.java

```diff
@@ -88,18 +89,22 @@
 
     @Override
     public void start() {
-        if (cczSelectionEnabled) {
-            setCczSelectionVisibility(true);
-        }
+        try {
+            if (cczSelectionEnabled) {
+                setCczSelectionVisibility(true);
+            }
 
-        if (mLastExecutionStatus == STATUS_FAILED) {
-            mActivity.enableRetry();
-            updateStatus(mLastDisplayStatus);
-        } else if (!(connectToUpdateTask() || mActivity.aTaskInProgress()) && mLastExecutionStatus != STATUS_WAITING) {
```

Comment:
Removed `mLastExecutionStatus != STATUS_WAITING` condition here since in case we don't have an exisiting task to connect to and mLastExecutionStatus is `STATUS_WAITING` means that we have lost result of the last async task. So the only resonable thing to do in this case is to trigger the measures again.